testutil/fakestorage/fixtures.go

```diff
@@ -40,46 +36,53 @@ func ensureHTTPServerImage(t testing.TB) {
 		goarch = "amd64"
 	}
 
-	cpCmd, lookErr := exec.LookPath("cp")
-	if lookErr != nil {
-		t.Fatalf("could not build http server: %v", lookErr)
-	}
+	goCmd, err := exec.LookPath("go")
+	assert.NilError(t, err, "could not find go executable to build http server")
 
-	if _, err := os.Stat("../contrib/httpserver/httpserver"); os.IsNotExist(err) {
-		goCmd, lookErr := exec.LookPath("go")
-		if lookErr != nil {
-			t.Fatalf("could not build http server: %v", lookErr)
-		}
+	tmp := t.TempDir()
+	const httpServer = `package main
 
-		cmd := exec.Command(goCmd, "build", "-o", filepath.Join(tmp, "httpserver"), "github.com/moby/moby/v2/contrib/httpserver")
-		cmd.Env = append(os.Environ(), []string{
-			"CGO_ENABLED=0",
-			"GOOS=" + goos,
-			"GOARCH=" + goarch,
-		}...)
-		var out []byte
-		if out, err = cmd.CombinedOutput(); err != nil {
-			t.Fatalf("could not build http server: %s", string(out))
-		}
-	} else {
-		if out, err := exec.Command(cpCmd, "../contrib/httpserver/httpserver", filepath.Join(tmp, "httpserver")).CombinedOutput(); err != nil {
-			t.Fatalf("could not copy http server: %v", string(out))
-		}
-	}
+import (
+	"log"
+	"net/http"
+)
 
-	if out, err := exec.Command(cpCmd, "../contrib/httpserver/Dockerfile", filepath.Join(tmp, "Dockerfile")).CombinedOutput(); err != nil {
-		t.Fatalf("could not build http server: %v", string(out))
-	}
+func main() {
+	fs := http.FileServer(http.Dir("/static"))
+	http.Handle("/", fs)
+	log.Panic(http.ListenAndServe(":80", nil)) // #nosec G114 -- Ignoring for test-code: G114: Use of net/http serve function that has no support for setting timeouts (gosec)
+}
+`
```

Comment:
It's slightly ugly to put this in a const, but with the current handling of "create a temporary build-context _somewhere_, add some other files, then do a build", it was easier to do a `os.WriteFile()` with the source then to move or copy files around.

We should probably have another look at this package and logic used in future.